pkg/endpoint/endpoint.go

```diff
@@ -2403,3 +2433,61 @@ func (e *Endpoint) IPs() []net.IP {
 func (e *Endpoint) InsertEvent() {
 	e.getLogger().Info("New endpoint")
 }
+
+// must be called with endpoint mutex locked.
+func (e *Endpoint) syncPolicyMap() error {
+
+	if e.realizedMapState == nil {
+		e.realizedMapState = make(map[policymap.PolicyKey]struct{})
+	}
+
+	currentMapContents, err := e.PolicyMap.DumpToSlice()
+
+	if err != nil {
+		return fmt.Errorf("unable to dump PolicyMap for endpoint: %s", err)
+	}
+
+	for _, entry := range currentMapContents {
+		keyToAnalyze := policymap.PolicyKey{
+			Identity:         entry.Key.GetIdentity(),
+			DestPort:         byteorder.NetworkToHost(entry.Key.GetPort()).(uint16),
+			Nexthdr:          entry.Key.GetProto(),
+			TrafficDirection: entry.Key.GetDirection(),
+		}
```

Comment:
Does this package need to deal with the details of the contents of `PolicyKey`? Could we use something like this instead?

`keyToAnalyze := entry.(policymap.PolicyKey)`

Reply:
I could add a helper function that converts the key to a "userspace friendly" format in the policymap package for this case?

Reply:
I think the only place it becomes really relevant is when you're trying to prettyprint the entries? Can we just always store them in NBO and convert to HBO in String() functions?

Reply:
@joestringer @raybejjani both good points. Will fix in next upload :)